File: TcpSocketSystem_DotNet/src/Domain/Puertos/IRepositorioSesiones.cs
using TcpSocketSystem.Core.Domain.Modelos;

namespace TcpSocketSystem.Core.Domain.Puertos;

/// <summary>
/// Contrato para la persistencia en memoria y consulta de sesiones de sockets activas.
/// </summary>
public interface IRepositorioSesiones
{
    void RegistrarSesion(EstadoSesion sesion);
    void ActualizarSesion(EstadoSesion sesion);
    void FinalizarSesion(string sesionId);
    EstadoSesion? ObtenerPorId(string sesionId);
    IReadOnlyCollection<EstadoSesion> ObtenerSesionesActivas();
    int ObtenerTotalSesionesActivas();
}

File: TcpSocketSystem_Java/src/com/tecnm/sockets/domain/puertos/IRepositorioSesiones.java
package com.tecnm.sockets.domain.puertos;

import com.tecnm.sockets.domain.modelos.EstadoSesion;
import java.util.Collection;

/**
 * Contrato para el repositorio de sesiones concurrentes.
 */
public interface IRepositorioSesiones {

    void registrarSesion(EstadoSesion sesion);

    void actualizarSesion(EstadoSesion sesion);

    void finalizarSesion(String sesionId);

    EstadoSesion obtenerPorId(String sesionId);

    Collection<EstadoSesion> obtenerSesionesActivas();

    int obtenerTotalSesionesActivas();
}
